data/liveinst/liveinst

```diff
@@ -20,11 +20,24 @@
 
 BACKEND_READY_FLAG=/run/anaconda/backend_ready
 
+WAYLAND_DISPLAY_SOCKET=/tmp/anaconda-wldisplay
+
+# Detect and save the wayland socket before re-exec
+if [ -n "$WAYLAND_DISPLAY" ]; then
+    if [ -e "$WAYLAND_DISPLAY_SOCKET" ]; then
+        rm -f "${WAYLAND_DISPLAY_SOCKET}"
+    fi
+    touch "${WAYLAND_DISPLAY_SOCKET}"
+    echo "${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY}" > ${WAYLAND_DISPLAY_SOCKET}
+fi
+
 # The command needs to be run with root privileges, so if it was started
 # unprivileged, restart running as root.
 if [ "$(id -u)" -ne 0 ]; then
-    xhost +si:localuser:root
-    unset XAUTHORITY
+    if [ -z "$WAYLAND_DISPLAY" ]; then
+        xhost +si:localuser:root
+        unset XAUTHORITY
+    fi
     pkexec "$0" "$@"
```

Comment:
If we do: `pkexec env WAYLAND_DISPLAY=$WAYLAND_DISPLAY "$0" "$@"` would the variable be retained ?

Reply:
Yes it would, but I would rather not do that because it confuses polkit (you're now triggering on `env` rather than `liveinst`)